lib/deimos/utils/db_poller/base.rb

```diff
@@ -21,17 +21,37 @@ class Base
         # @return [Hash]
         attr_reader :config
 
+        # Method to define producers if a single poller needs to publish to multiple topics.
+        # Producer classes should be constantized
+        # @return [Array<Producer>]
+        def self.producers
+          []
+        end
+
         # @param config [FigTree::ConfigStruct]
         def initialize(config)
           @config = config
           @id = SecureRandom.hex
           begin
-            @producer = @config.producer_class.constantize
-          rescue NameError
-            raise "Class #{@config.producer_class} not found!"
-          end
-          unless @producer < Deimos::ActiveRecordProducer
-            raise "Class #{@producer.class.name} is not an ActiveRecordProducer!"
+            if @config.poller_class.nil? && @config.producer_class.nil?
+              raise 'No producers have been set for this DB poller!'
+            end
+
+            @resource_class = @config.producer_class.present? ? @config.producer_class.constantize : @config.poller_class.constantize
+
+            if @config.poller_class.nil?
+              unless @resource_class < Deimos::ActiveRecordProducer
+                raise "Class #{@resource_class.class.name} is not an ActiveRecordProducer!"
+              end
+            else
+              @resource_class.producers.each do |producer|
+                unless producer < Deimos::ActiveRecordProducer
+                  raise "Class #{producer.class.name} is not an ActiveRecordProducer!"
+                end
+              end
+            end
+          rescue NameError => e
+            raise e.message.to_s
```

Comment:
Is there a reason we're just re-raising the error here?

Reply:
I guess the reason I changed it was because before it was `raise "Class #{@config.producer_class} not found!"`, but now it could either be `@config.producer_class` or `@config.poller_class`, and if it fails when one of them get `constantize` called on it, I guess we would need to check which one is populated since at this stage if it fails, `@resource_class` wouldn't have even got populated.

Reply:
If we fix having to deal with the `if` statements, this becomes a lot easier :) You can pass whatever class to the validation method and have it handle this.

Reply:
Sounds good 👍